### PR TITLE
Add Allure upload command support

### PR DESCRIPTION
**Pull request by satvik007**
## Summary

Adds a new `allure-upload` command that ingests Allure 2 results directories and uploads them to QA Sphere.

### Allure parser (`allureParser.ts`)
- Reads `*-result.json` files from a directory; ignores containers, XML, and raw attachments
- Test case matching via TMS links (`type: "tms"`) or marker in test name (e.g. `BD-123`)
- Status mapping: `passed → passed`, `failed → failed`, `broken → blocked`, `skipped → skipped`, `unknown → open`
- Resolves attachments by `source` filename within the same directory; also collects attachments from nested steps recursively
- Strips ANSI codes and HTML-escapes messages/stack traces
- Derives suite/folder from Allure `suite`/`parentSuite`/`subSuite` labels
- Uses `formatMarker()` from `MarkerParser` for consistent marker formatting
- `allowPartialParse` support: skips invalid result files instead of aborting when `--force` is set

### Command wiring
- Registers `allure-upload` as a third command in `main.ts` via the same `ResultUploadCommandModule`
- Added `allure-upload` to `directoryInputTypes` in `ResultUploadCommandHandler` — the parser receives the directory path rather than file contents
- Unmatched-test error guidance now includes allure-specific hints
- `getMarkerFromText` added to `MarkerParser` and exposed in tests

### Tests & fixtures
- `allure-parsing.spec.ts` — 545-line unit test suite covering TMS links, markers, status mapping, suite labels, attachment resolution, nested step attachments, partial parse, invalid files, and edge cases
- `result-upload.spec.ts` — integration tests for the full allure upload flow via MSW
- Fixture directories: `matching-tcases`, `missing-tcases`, `empty-tsuite`, `invalid-results`, `missing-attachments`, `without-markers`

### Documentation
- README updated with `allure-upload` usage, options, and examples

## Testing

- [x] `npm test` — all unit and integration tests pass
- [x] `npm run check` — typecheck + lint + format
- [x] Manual smoke test against bistro-e2e Allure reports: uploaded 6 test results with 6 PNG screenshot attachments to a live QA Sphere instance — clean run, no errors

Resolves: https://github.com/Hypersequent/tms-issues/issues/2329

### CODE REVIEW

File: src/commands/resultUpload.ts [hunk 1]
```diff
@@ -10,11 +10,19 @@ import {
 const commandTypeDisplayStrings: Record<UploadCommandType, string> = {
 	'junit-upload': 'JUnit XML',
 	'playwright-json-upload': 'Playwright JSON',
+	'allure-upload': 'Allure',
 }
 
-const commandTypeFileExtensions: Record<UploadCommandType, string> = {
-	'junit-upload': 'xml',
-	'playwright-json-upload': 'json',
+const commandTypeInputKinds: Record<UploadCommandType, string> = {
```
Comment (hi-rai) [inline src/commands/resultUpload.ts]: nit: We can now have a single `commandDetails` map - `Record<UploadCommandType, CommandDetails>`, where `CommandDetails` is an object holding all the (three) required information about the commands. And in the constructor, we can do `this.commandDetails = commandDetails[type]`
